From 369140ab89948265c722a9b7a7e92b7de4a88d02 Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Fri, 28 Aug 2020 10:51:28 +0930
Subject: [PATCH 2/3] mingw plugin test regressions due to commit 514b4e191d5f

Fixes new failures due to image base change.

	PR 19011
	* testsuite/ld-plugin/plugin.exp: Use modified CFLAGS throughout
	file.  Add --image-base for pecoff.
---
 ld/ChangeLog                      | 6 ++++++
 ld/testsuite/ld-plugin/plugin.exp | 8 ++++++++
 2 files changed, 14 insertions(+)

diff --git a/ld/ChangeLog b/ld/ChangeLog
index 0a65318c46..b952c547db 100644
--- a/ld/ChangeLog
+++ b/ld/ChangeLog
@@ -1,3 +1,9 @@
+2020-08-28  Alan Modra  <amodra@gmail.com>
+
+	PR 19011
+	* testsuite/ld-plugin/plugin.exp: Use modified CFLAGS throughout
+	file.  Add --image-base for pecoff.
+
 2020-08-27  Jeremy Drake  <sourceware-bugzilla@jdrake.com>
 
 	PR 19011
diff --git a/ld/testsuite/ld-plugin/plugin.exp b/ld/testsuite/ld-plugin/plugin.exp
index dc8f007397..bbff5d9d55 100644
--- a/ld/testsuite/ld-plugin/plugin.exp
+++ b/ld/testsuite/ld-plugin/plugin.exp
@@ -95,6 +95,7 @@ set _ ""
 set plugin_nm_output ""
 set old_CFLAGS "$CFLAGS"
 append CFLAGS " $NOSANTIZE_CFLAGS"
+
 if { $can_compile && \
 	(![ld_compile "$CC $CFLAGS" $srcdir/$subdir/main.c tmpdir/main.o] \
 	|| ![ld_compile "$CC $CFLAGS" $srcdir/$subdir/func.c tmpdir/func.o] \
@@ -129,6 +130,10 @@ set libs "[ld_link_defsyms] --defsym ${_}printf=${_}main --defsym ${_}puts=${_}m
 if { $dotsym } {
     append libs " --defsym .printf=.main --defsym .puts=.main"
 }
+if [is_pecoff_format] {
+    #otherwise relocs overflow to symbols defined on the command line
+    append libs " --image-base=0x10000000"
+}
 
 set plugin_tests [list \
     [list "load plugin" "-plugin $plugin_path \
@@ -289,6 +294,7 @@ if { !$can_compile || $failed_compile } {
 	    $failure_kind [lindex $testitem 0]
 	}
     }
+    set CFLAGS "$old_CFLAGS"
     return
 }
 
@@ -390,3 +396,5 @@ if [ar_simple_create $ar "--plugin $plugin4_path" "tmpdir/libpr20070.a" \
 } else {
     unresolved "PR ld/20070"
 }
+
+set CFLAGS "$old_CFLAGS"
-- 
2.28.0.windows.1

